src/DisputeHandler.ts

```diff
@@ -55,164 +72,281 @@ class DisputeHandler {
+    } {
+        const timeout: TimeoutStruct = {
+            participant: ethers.ZeroAddress,
+            blockHeight: ethers.MaxUint256,
+            minTimeStamp: ethers.MaxUint256,
+            forkCnt: 0,
+            isForced: false,
+            previousBlockProducer: ethers.ZeroAddress,
+            previousBlockProducerPostedCalldata: false
+        };
+
+        let dispute: DisputeStruct = {
+            channelId: this.channelId,
+            genesisStateSnapshotHash: "",
+            latestStateSnapshotHash: "",
+            stateProof: {
+                forkProof: {
+                    forkMilestoneProofs: []
+                },
+                signedBlocks: []
+            },
+            fraudProofs: [],
+            onChainSlashes: [],
+            onChainLatestJoinChannelBlockHash: "",
+            outputStateSnapshotHash: "",
+            exitChannelBlocks: [],
+            disputeAuditingDataHash: "",
+            disputer: this.signerAddress,
+            disputeIndex: 0,
+            previousRecursiveDisputeIndex: ethers.MaxUint256,
+            timeout,
+            selfRemoval: false
+        };
+        return { dispute, timeout };
     }
 
-    //Creates a dispute based on the generated proofs or optimistically timeouts (folds) the provided participant
-    public async createDispute(
-        forkCnt: BigNumberish,
-        foldedParticipant: AddressLike,
-        foldedTransactionCnt: BigNumberish,
-        proofs: ProofStruct[]
-    ): Promise<void> {
-        if (foldedParticipant != NO_PARTICIPANT_TO_FOLD) {
-            console.log("DisputeHandler - createDispute - Timeout");
-        }
+    private async createDisputeStruct(
+        forkCnt: number,
+        transactionCnt: number,
+        proofs: ProofStruct[],
+        timestamp: BigNumberish,
+        timeout?: TimeoutStruct
+    ): Promise<{
+        dispute: DisputeStruct;
+        disputeAuditingData: DisputeAuditingDataStruct;
+    }> {
+        const disputeIndex =
+            await this.stateChannelManagerContract.getDisputeLength(
+                this.channelId
+            );
+        const genesisStateSnapshotHash =
+            this.agreementManager.getForkGenesisSnapshot(
+                forkCnt
+            )!.stateMachineStateHash;
+        const latestStateSnapshotHash = this.agreementManager.getSnapShot(
+            forkCnt,
+            transactionCnt
+        )!.stateMachineStateHash;
+        const stateProof = this.collectStateProof(forkCnt);
+        const onChainSlashes =
+            await this.stateChannelManagerContract.getOnChainSlashedParticipants(
+                this.channelId
+            );
+        const onChainLatestJoinChannelBlockHash =
+            this.agreementManager.getLatestJoinChannelBlockHash();
+        const latestExitChannelBlockHash =
+            this.agreementManager.getLatestJoinChannelBlockHash();
+        const exitChannelBlocks = this.agreementManager.getExitChannelChain();
```

Comment:
This should not be the entire blockchain, but only exits that happened within the fork (since genesis till the latest)